rust/query-stats/src/lib.rs

```diff
@@ -0,0 +1,207 @@
+use candid::{CandidType, Nat};
+use ic_cdk::api::management_canister::main::canister_status;
+use ic_cdk::api::management_canister::provisional::CanisterIdRecord;
+
+use ic_cdk::api::management_canister::main::QueryStats;
+
+use std::cell::RefCell;
+use std::time::Duration;
+
+const TIMER_PERIOD_IN_SECS: u64 = 5;
+const QUERY_STATS_PURGE_AFTER_SECS: u64 = 60 * 60 * 24; // 24h, has to be much larger than query stats epoch
+const QUERY_STATS_RATE_FOR_LAST_SECS: u64 = 60 * 60; // 1h, has to be larger than query stats epoch
+
+// Unfortunately, we need to redefine this type, since the members are not public.
```

Comment:
```suggestion
// Unfortunately, we need to redefine this type, since the members are not public.
```
Can this be fixed somehow? E.g. by including it in some appropriate crate?